example/cosmos/main.go

```diff
@@ -0,0 +1,171 @@
+// (c) Copyright IBM Corp. 2024
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/data/azcosmos"
+	"github.com/google/uuid"
+	instana "github.com/instana/go-sensor"
+	"github.com/instana/go-sensor/instrumentation/instacosmos"
+)
+
+const (
+	database  = "trace-data"
+	container = "spans"
+)
+
+const (
+	CONNECTION_URL = "COSMOS_CONNECTION_URL"
+	KEY            = "COSMOS_KEY"
+)
+
+type SpanType string
+
+const (
+	EntrySpan SpanType = "entry"
+	ExitSpan  SpanType = "exit"
+)
+
+type Span struct {
+	ID          string   `json:"id"`
+	SpanID      string   `json:"SpanID"`
+	Type        SpanType `json:"type"`
+	Description string   `json:"description"`
+}
+
+var (
+	collector instana.TracerLogger
+)
+
+var (
+	endpoint = ""
+	key      = ""
+)
+
+func init() {
+	validateAzureCreds()
+	collector = instana.InitCollector(&instana.Options{
+		Service: "sample-app-cosmos",
+	})
+}
+
+func main() {
+	http.HandleFunc("/cosmos-test", instana.TracingHandlerFunc(collector, "/cosmos-test", handler))
+	log.Fatal(http.ListenAndServe("localhost:9990", nil))
+}
+
+func handler(w http.ResponseWriter, r *http.Request) {
+
+	erStr := r.URL.Query().Get("error")
+	needError := erStr == "true"
+	if needError {
```

Comment:
I think it's safe to remove this if statement. We are already setting needError to either true or false on line 70 :)

Reply:
Oh yes. removed it 👍